services/server/test/integration/verification-handlers/verify.stateless.spec.ts

```diff
@@ -172,7 +174,47 @@ describe("/", function () {
       );
   });
 
-  it("Should upgrade match from 'partial' to 'full', delete partial from repository and update creationTx information in database", async () => {
+  it("Should skip compilation if contract is already verified", async () => {
+    let res = await chai
+      .request(serverFixture.server.app)
+      .post("/")
+      .field("address", chainFixture.defaultContractAddress)
+      .field("chain", chainFixture.chainId)
+      .attach("files", chainFixture.defaultContractMetadata, "metadata.json")
+      .field("creatorTxHash", chainFixture.defaultContractCreatorTx)
+      .attach("files", chainFixture.defaultContractSource);
+    await assertVerification(
+      serverFixture.sourcifyDatabase,
+      null,
+      res,
+      null,
+      chainFixture.defaultContractAddress,
+      chainFixture.chainId,
+      "perfect",
+    );
+    // The first time the contract is verified, the storageTimestamp is not returned
+    chai.expect(res.body.result[0].storageTimestamp).to.not.exist;
+    res = await chai
+      .request(serverFixture.server.app)
+      .post("/")
+      .field("address", chainFixture.defaultContractAddress)
+      .field("chain", chainFixture.chainId)
+      .attach("files", chainFixture.defaultContractMetadata, "metadata.json")
+      .attach("files", chainFixture.defaultContractSource);
+    await assertVerification(
+      serverFixture.sourcifyDatabase,
+      null,
+      res,
+      null,
+      chainFixture.defaultContractAddress,
+      chainFixture.chainId,
+      "perfect",
+    );
+    // If the contract is perfectly verified (both creation and runtime), the storageTimestamp is returned
+    chai.expect(res.body.result[0].storageTimestamp).to.exist;
```

Comment:
I am not sure if this is the right way to check if compilation was skipped. This test is based on the assumption that the storageTimestamp is given in such a situation. But I would rather explicitly check that the compiler is not called. You can do that with sinon.js (I added it in a PR which then was not merged). A spy is the right concept for this I think.